src/main.cpp

```diff
@@ -1,31 +1,65 @@
 #include "server/Server.hpp"
 #include "config/ConfigParser.hpp"
+#include "config/ConfigFactory.hpp"
 
 #include <iostream>
 
 static bool	isParsingFlag(int argc, char *argv[]);
-static void	parseConfigFile(const std::string& configFile);
+static bool parseConfigFile(const ConfigLexer& lexer);
+static int 	testConfigFile(const std::string& configFile);
+static int	runServer(const std::string& configFile);
 
 int main(int argc, char *argv[])
 {
-	if (isParsingFlag(argc, argv))
-		parseConfigFile(argv[2]);
-	else {
-		Server	webserv = Server();
-		webserv.listen();
+	if (!isParsingFlag(argc, argv) && argc != 2) {
+		std::cerr << "Usage: ./webserv *.conf" << std::endl;
+		return EXIT_FAILURE;
 	}
+	if (isParsingFlag(argc, argv))
+		return testConfigFile(argv[2]);
+	else
+		return runServer(argv[1]);
 	return EXIT_SUCCESS;
 }
 
 static bool	isParsingFlag(int argc, char *argv[]) {
 	return (argc == 3 && std::string(argv[1]) == "-t");
 }
 
-static void	parseConfigFile(const std::string& configFile) {
+static int	testConfigFile(const std::string& configFile) {
+	ConfigLexer	lexer(configFile);
+	if (!parseConfigFile(lexer))
+		return EXIT_FAILURE;
+	std::cout << "Syntax is ok" << std::endl;
+	return EXIT_SUCCESS;
+}
+
+static int runServer(const std::string& configFile) {
+	ConfigLexer*	lexer = new ConfigLexer(configFile);
```

Comment:
Add a scope in order to allocate the lexer in the stack memory :
static int runServer(const std::string& configFile) {
  Server webserv;
  {
    lexer(webserv);
  }
  
  webserv.run()
}